dashboard/src/components/Config/AppRepoList/AppRepoForm.tsx

```diff
@@ -0,0 +1,259 @@
+import * as React from "react";
+import { Redirect } from "react-router";
+
+interface IAppRepoFormProps {
+  message?: string;
+  redirectTo?: string;
+  install: (name: string, url: string, authHeader: string, customCA: string) => Promise<boolean>;
+  onAfterInstall?: () => Promise<any>;
+}
+
+interface IAppRepoFormState {
+  useRawAuthHeader: boolean;
+  useBasicAuth: boolean;
+  useBearerToken: boolean;
+  user: string;
+  password: string;
+  authHeader: string;
+  token: string;
+  name: string;
+  url: string;
+  customCA: string;
+}
+
+export class AppRepoForm extends React.Component<IAppRepoFormProps, IAppRepoFormState> {
+  public state = {
+    useRawAuthHeader: false,
+    useBasicAuth: false,
+    useBearerToken: false,
+    user: "",
+    password: "",
+    authHeader: "",
+    token: "",
+    name: "",
+    url: "",
+    customCA: "",
+  };
+
+  public render() {
+    return (
+      <form className="container padding-b-bigger" onSubmit={this.handleInstallClick}>
+        <div className="row">
+          <div className="col-12">
+            <div>
+              <h2>Add an App Repository</h2>
+            </div>
+            <div>
+              <label>
+                <span>Name:</span>
+                <input
+                  type="text"
+                  placeholder="example"
+                  value={this.state.name}
+                  onChange={this.handleNameChange}
+                  required={true}
+                  pattern="[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*"
+                  title="Use lower case alphanumeric characters, '-' or '.'"
+                />
+              </label>
+            </div>
+            <div>
+              <label>
+                <span>URL:</span>
+                <input
+                  type="url"
+                  placeholder="https://charts.example.com/stable"
+                  value={this.state.url}
+                  onChange={this.handleURLChange}
+                  required={true}
+                />
+              </label>
+            </div>
+            <div>
+              <label>
+                <span>Authorization (optional):</span>
+                <div className="row">
+                  <div className="col-3">
+                    <label className="margin-l-big">
+                      <input
+                        type="radio"
+                        name="auth"
+                        value="None"
+                        defaultChecked={true}
+                        onChange={this.handleAuthRadioButtonChange}
+                      />
+                      None
+                      <br />
+                    </label>
+                    <label>
```

Comment:
Set the value of the `htmlFor` property to the `id` of its input. This will connect both for better accessibility and allow users to change the radio when clicking on the text